src/ZODB/Connection.py

```diff
@@ -1321,3 +1321,22 @@ def _extension(self):
     @_extension.setter
     def _extension(self, v):
         self.extension = v
+
+    def data(self, ob):
+        try:
+            data = self._data
+        except AttributeError:
+            raise KeyError(ob)
+
+        try:
+            return data[id(ob)]
+        except KeyError:
+            raise KeyError(ob)
```

Comment:
Is this simpler and equivalent (fewer lines of code to cover)?
```python
try:
   return self._data[id(ob)]
except (AttributeError, KeyError):
   raise KeyError(ob)
```

Not that it's a big deal. If you prefer the more explicit version that's fine by me.